utils/build-script-impl

```diff
@@ -2012,7 +2012,6 @@ for host in "${ALL_HOSTS[@]}"; do
                     -DCLANG_TOOL_ARCMT_TEST_BUILD=NO
                     -DCLANG_TOOL_C_ARCMT_TEST_BUILD=NO
                     -DCLANG_TOOL_C_INDEX_TEST_BUILD=NO
-                    -DCLANG_TOOL_DRIVER_BUILD=NO
```

Comment:
`BUILD_TOOLCHAIN_ONLY` should do a lot of things differently when cross compiling, because right now it does a lot of extra work. If you turn on `BUILD_TOOLCHAIN_ONLY` today the project won't build at all because `clang` is missing. `BUILD_TOOLCHAIN_ONLY` also is only about improving build times, so correctness comes first.

We should definitely investigate removing table gen, the clang frontend and other tools that run during the build for a cross compile. But that is something we should investigate after fixing the build.